src/main/scala/com/exasol/cloudetl/bucket/Bucket.scala

```diff
@@ -64,8 +70,50 @@ abstract class Bucket extends LazyLogging {
    *
    * This method also globifies the bucket path if it contains regex.
    */
-  final def getPaths(): Seq[Path] =
-    FileSystemUtil.globWithPattern(bucketPath, fileSystem)
+  final def getPaths(): Seq[Path] = properties.getFileFormat() match {
+    case FileFormat.DELTA => getPathsFromDeltaLog()
+    case _                => FileSystemUtil.globWithPattern(bucketPath, fileSystem)
+  }
+
+  private[this] def getPathsFromDeltaLog(): Seq[Path] = {
+    val spark = createSparkSession()
+    val strippedBucketPath = stripTrailingStar(bucketPath)
+    val deltaLog = DeltaLog.forTable(spark, strippedBucketPath)
+    if (!deltaLog.isValid()) {
+      throw new IllegalArgumentException(
+        s"The provided path: '$bucketPath' is not a Delta format!"
+      )
+    }
+    val latestSnapshot = deltaLog.update()
+
+    latestSnapshot.allFiles
+      .select("path")
+      .collect()
+      .map { case Row(path: String) => new Path(s"$strippedBucketPath/$path") }
+  }
+
+  private[this] def createSparkSession(): SparkSession = {
+    lazy val spark = SparkSession
+      .builder()
+      .master("local[*]")
+      .config("spark.delta.logStore.class", properties.getDeltaFormatLogStoreClassName())
+      .getOrCreate()
+
+    getConfiguration().iterator().asScala.foreach { entry =>
+      val key = entry.getKey()
+      val value = entry.getValue()
```

Comment:
What do the key and value represent? It's not clear from names

Reply:
It is a [HadoopConfiguration](https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/conf/Configuration.html), that stores configuration parameters as `Map.Entry[String, String]`. 
Going to rename them to `hadoopConfigKey/Value`.